index.js

```diff
@@ -1,121 +1,118 @@
         }
-
-        return '(' + minmax[gtlt] + '-' + name + ': ' + number + unit + ')';
-      });
-    }
-
-    // 读取 media-feature
-    css.walkAtRules(function(rule, i) {
-      if (rule.name !== "media" && rule.name !== "custom-media") {
-        return
       }
+    } else {
+      number = power[gtlt] + feature_unit[name];
+    }
 
-      /**
-       * 转换 <mf-name> <|>= <mf-value>
-       *    $1  $2   $3
-       * (width >= 300px) => (min-width: 300px)
-       * (width <= 900px) => (max-width: 900px)
-       */
-
-      //取值不支持负值
-      //But -0 is always equivalent to 0 in CSS, and so is also accepted as a valid <mq-boolean> value.
+    return '(' + minmax[gtlt] + '-' + name + ': ' + number + unit + ')';
+  });
+}
 
-      rule.params = rule.params.replace(/\(\s*([a-z-]+?)\s*([<>])(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4) {
+function transform(rule) {
+  /**
+   * 转换 <mf-name> <|>= <mf-value>
+   *    $1  $2   $3
+   * (width >= 300px) => (min-width: 300px)
+   * (width <= 900px) => (max-width: 900px)
+   */
 
-        var params = '';
+  // The value doesn't support negative values
+  // But -0 is always equivalent to 0 in CSS, and so is also accepted as a valid <mq-boolean> value.
 
-        if (feature_name.indexOf($1) > -1) {
-          return create_query($1, $2, $3, $4, rule.params);
-        }
-        //如果不是指定的属性，不做替换
-        return $0;
-      })
-
-      /**
-       * 转换  <mf-value> <|<= <mf-name> <|<= <mf-value>
-       * 转换  <mf-value> >|>= <mf-name> >|>= <mf-value>
-       *   $1  $2$3 $4  $5$6  $7
-       * (500px <= width <= 1200px) => (min-width: 500px) and (max-width: 1200px)
-       * (500px < width <= 1200px) => (min-width: 501px) and (max-width: 1200px)
-       * (900px >= width >= 300px)  => (min-width: 300px) and (max-width: 900px)
-       */
-
-      rule.params = rule.params.replace(/\(\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*(<|>)(=?)\s*([a-z-]+)\s*(<|>)(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4, $5, $6, $7) {
-
-        if (feature_name.indexOf($4) > -1) {
-          if ($2 === '<' && $5 === '<' || $2 === '>' && $5 === '>') {
-            var min = ($2 === '<') ? $1 : $7;
-            var max = ($2 === '<') ? $7 : $1;
-
-            // output differently depended on expression direction
-            // <mf-value> <|<= <mf-name> <|<= <mf-value>
-            // or
-            // <mf-value> >|>= <mf-name> >|>= <mf-value>
-            var equals_for_min = $3;
-            var equals_for_max = $6;
-
-            if ($2 === '>') {
-              equals_for_min = $6;
-              equals_for_max = $3;
-            }
-
-            return create_query($4, '>', equals_for_min, min) + ' and ' + create_query($4, '<', equals_for_max, max);
-          }
+  rule.params = rule.params.replace(/\(\s*([a-z-]+?)\s*([<>])(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4) {
```

Comment:
```suggestion
  if (!atRule.params.includes('<') && !atRule.params.includes('>')) return

  rule.params = rule.params.replace(/\(\s*([a-z-]+?)\s*([<>])(=?)\s*((?:-?\d*\.?(?:\s*\/?\s*)?\d+[a-z]*)?)\s*\)/gi, function($0, $1, $2, $3, $4) {
```

A quick check will avoid calling the heavy parser for at-rules without any `<` and `>`.

Reply:
Thanks!

Reply:
Updated